File: leetcode/problems/c/missing-number.c
// by @codeAbinash
// Time Complexity : O(n)
// Space Complexity : O(1)

// The missing number is (seriesSum - arraySum)
int missingNumber(int* nums, int numsSize){
    int sum = 0;
    for(int i = 0; i < numsSize; i++)
        sum+=nums[i];
    int seriesSum = (numsSize * (numsSize + 1) / 2);
    return seriesSum - sum;
}

File: leetcode/problems/java/missing-number.java
class Solution {
  public int missingNumber(int[] nums) {
    int n = nums.length;
    int xor1 = 0, xor2 = 0;
    for (int i = 0; i < n; i++) {
      xor1 ^= nums[i];
      xor2 ^= i;
    }
    xor2 ^= n;
    return xor1 ^ xor2;
  }
}